### ISSUE TITLE
Define some keywords for the primary hdu

### ISSUE THREAD

**Issue by MaxNoe** [open-gamma-ray-astro/gamma-astro-data-formats#141]
I think now, we do not make any definitions for the primary hdu.

Maybe we should also include at least some header keywords here so a file can be immediately recognized as a file produced according to theses standards here.

**Comment by lmohrmann** [comment]
Sounds like a good idea. Can you make a proposal?

**Comment by kosack** [comment]
Might consider just using the keywords that OFWG already recommends:

The OFWG recommends that the following set of keywords be used to for such purposes in each extension:

* HDUCLASS - a character string giving a general identifier of data format used
* HDUDOC - a character string giving the document (preferably published) that describes the format/classification used
* HDUVERS - a character string giving the specific version of the document specified by HDUDOC.
* HDUCLASn - an indexed set of character strings giving the classification of data in the extension

Current allowed values here: https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/ofwg_recomm/hduclas.html

You'd have to invent a HDUCLASS I guess.

**Comment by MaxNoe** [comment]
These are used for the actual HDUs already.

So I'd suggest HDUCLASS = `OGADF_PRIMARY`?

**Comment by TarekHC** [comment]
> These are used for the actual HDUs already.
> 
> So I'd suggest HDUCLASS = `OGADF_PRIMARY`?

I agree with this. 
 
>Maybe we should also include at least some header keywords here so a file can be immediately recognized as a file produced according to theses standards here. 

Perhaps we could add a list of the different tables within the DL3 file? Perhaps that is too complex?

**Comment by MaxNoe** [comment]
> Perhaps that is too complex?

Not really, I think.

Is there a something standard for this?
If not, something like this:

```
HDU0 = OGADF_PRIMARY
HDU1 = EVENTS
HDU2 = RESPONSE/EFF_AREA/POINT-LIKE/AEFF_2D
HDU3 = RESPONSE/EDISP/POINT-LIKE/EDISP_2D
```

This could be optional but recommended in the standard, since you can always get this information thorugh looking at the individual hdus's

**Comment by jknodlseder** [comment]
Here is the standard

https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/ofwg_recomm/hduclas.html

Note sure what the "_PRIMARY" stands for. "OGADF" for HDUCLASS seems sufficient.

**Comment by MaxNoe** [comment]
@jknodlseder this is about the first primary hdu, that is not used in this standard for anything right now.
`HDUCLASS` is recommended to be `GADF` in the current standard, so the primary hdu (the image  hdu the FITS standard requires  as first hdu) could have:

```
HDUCLASS = "GADF"
HDUDOC    = "https://..."
HDUVERS   = "v0.3"
HDUCLAS1 = "PRIMARY"
```

**Comment by TarekHC** [comment]
Going through the format used within the `CREF#` keywords in Fermi-LAT IRFs (also containing arrays), the way they store them is:

Instead of @MaxNoe proposal:

> HDU0 = OGADF_PRIMARY
> HDU1 = EVENTS
> HDU2 = RESPONSE/EFF_AREA/POINT-LIKE/AEFF_2D
> HDU3 = RESPONSE/EDISP/POINT-LIKE/EDISP_2D

It would be:
```
HDU0 = OGADF_PRIMARY
HDU1 = EVENTS
HDU2 = (RESPONSE,EFF_AREA,POINT-LIKE,AEFF_2D)
HDU3 = (RESPONSE,EDISP,POINT-LIKE,EDISP_2D)
```

Maybe the initial "(" helps identifying that an array is coming? Anyway, if we can be as consistent as possible with other approaches used in the past, better.

**Comment by MaxNoe** [comment]
That's actually not standard conform, so we should not use it.

Enclosed in quotes as a string, it would be.

**Comment by jknodlseder** [comment]
Using HDUCLASS for an empty extension does not make any sense. HDUCLASS is meant to identify content.

**Comment by kosack** [comment]
I agree with Jurgen. You don't need an HDUCLASS in the primary.  Couldn't you just identify the file type by the HDUs present?   There's also the hierarchical grouping standard to group a set of HDUs together, but I think it also leave the primary HDU blank.  https://fits.gsfc.nasa.gov/registry/grouping.html
